File: VB/第二套/11900-1060307/Program.vb
Imports System.Windows.Forms

Module Program
    <STAThread()>
    Sub Main()
        Application.EnableVisualStyles()
        Application.SetCompatibleTextRenderingDefault(False)

        Dim view As New Views.MainForm()
        Dim poker As New Services.PokerService()
        Dim data As New Services.DataService()
        
        Dim presenter As New Presenters.MainPresenter(view, poker, data)
        
        Application.Run(view)
    End Sub
End Module

File: C#/第二套/11900-1060307/Program.cs
using Microsoft.Extensions.DependencyInjection;
using PokerGame.Views;
using PokerGame.Presenters;
using PokerGame.Services;

namespace PokerGame;

static class Program
{
    [STAThread]
    static void Main()
    {
        ApplicationConfiguration.Initialize();

        var services = new ServiceCollection();

        // Register Services
        services.AddSingleton<IDataService, DataService>();
        services.AddSingleton<IPokerService, PokerService>();

        // Register Views
        services.AddSingleton<IMainView, MainForm>();

        // Register Presenters
        services.AddSingleton<MainPresenter>();

        using var serviceProvider = services.BuildServiceProvider();

        // Resolve Presenter
        var presenter = serviceProvider.GetRequiredService<MainPresenter>();
        var mainForm = (Form)serviceProvider.GetRequiredService<IMainView>();

        Application.Run(mainForm);
    }
}
